mautic/mautic
# Upgrade to Symfony 4

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 4.0
| Bug fix?                               | no
| New feature?                           | no
| Deprecations?                          | yes
| BC breaks?                             | yes
| Automated tests included?              | some
| Related user documentation PR URL      | n/a
| Related developer documentation PR URL | todo

#### Description:

This is a PR that upgrades Mautic to Symfony 4 and minimal changes to make it "work." Please note the UPGRADE-4.0.md doc. 

There are some notes in https://docs.google.com/document/d/1VNQ5cuXJdzsgxi-HTfZ_ihSVrNBSWZwgClG6pabY7nc/edit#heading=h.2v7yth9cwgwg. A formal project/plan will be coming soon. 

Changes so far:
* Upgrade Symfony 4 (required some "hacks" to make services public and auto-register Commands; preferable these get refactored but will track in the planning board)
* Upgraded to latest El Finder (required to remove abandoned packages)
* Requires https://github.com/mautic/BazingaOAuthServerBundle/tree/release-0.0.4 for OAuth1a compatibility
* Refactored ModeratedCommand (Symfony removed LockHandler)
* Refactored some things in PipeDrive integration
* Upgraded GuzzleHttp to v7
* Bumped minimal PHP version to 7.4 - we're planning to add support for PHP 8.0 later on

## How to test

It's easiest if you do a **clean checkout** of this PR, e.g. in a new folder:

1. `gh repo clone mautic/mautic && cd mautic && gh pr checkout 9409`
2. If you have DDEV, simply run `ddev start` to get up and running, it will do the installation for you as well!
3. Mautic should install without errors. You should be able to click around, but don't expect all features to work flawlessly just yet, we haven't tested them.
